### PR TITLE
Add docs for struct operations

### PR DESCRIPTION
**Pull request by avevad**
### Changelog entry 

Add docs for struct operations

### Changelog category 

* Documentation (changelog entry is not required)

### Additional information

...

### CODE REVIEW

File: ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md [hunk 1]
```diff
@@ -392,3 +392,39 @@ ForceSpreadMembers(list_of_tuples:List<Tuple<String, T>>, result_keys:List<Strin
 SELECT
   ForceSpreadMembers([('a',1),('a',2),('c',100)],['a','b']); -- (a: 2, b: null)
 ```
+
+## StructUnion, StructIntersection, StructDifference, StructSymmetricDifference
+
+Комбинируют две структуры одним из четырех способов, используя предоставленную функцию для слияния полей с одинаковыми именами.
```
Comment (blinkov) [inline ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md:398]:
```suggestion
Комбинируют две структуры одним из четырех способов, используя предоставленную функцию для слияния полей с одинаковыми именами:

* `StructUnion` добавляет в результат все поля обеих структур;
* `StructIntersection` — поля, которые есть в обеих структурах; 
* `StructDifference` — поля которые есть в left, но которых нет в right;
* `StructSymmetricDifference` — все поля, которые есть только в одной из структур.
```

File: ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md [hunk 2]
```diff
@@ -392,3 +392,39 @@ ForceSpreadMembers(list_of_tuples:List<Tuple<String, T>>, result_keys:List<Strin
...
+    StructUnion($left, $right),                        -- <|a: 1, b: 2, c: 3, d: 2, e: 3|>
+    StructUnion($left, $right, $merge),                -- <|a: 1, b: 2, c: 4, d: 2, e: 3|>
+    StructIntersection($left, $right, $merge),         -- <|c: 4|>
+    StructDifference($left, $right, $merge),           -- <|a: 1, b: 1|>
+    StructSymmetricDifference($left, $right, $merge)   -- <|a: 1, b: 2, d: 2, e: 3|>
+;
+```
+
+Union добавляет в результат все поля обеих структур; Intersection - поля, которые есть в обеих структурах; Difference - поля которые есть в left, но которых нет в right; SymmetricDifference - все поля, которые есть только в одной из структур.
+
```
Comment (blinkov) [inline ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md:430]:
```suggestion
```

File: ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md [hunk 3]
```diff
@@ -392,3 +392,39 @@ ForceSpreadMembers(list_of_tuples:List<Tuple<String, T>>, result_keys:List<Strin
...
+StructIntersection(left:Struct<...>, right:Struct<...>[, mergeLambda:(name:String, l:T1?, r:T2?)->T])->Struct<...>
+StructDifference(left:Struct<...>, right:Struct<...>)->Struct<...>
+StructSymmetricDifference(left:Struct<...>, right:Struct<...>)->Struct<...>
+```
+
+Аргументы:
+
+1. `left` - первая структура
+2. `right` - вторая структура
+3. `mergeLambda` - _(опционально)_ позволяет задать функцию для объединения полей (аргументы: имя поля, Optional значение поля в первой структуре, Optional значение поля во второй структуре); по умолчанию выбирается значение поля из первой структуры, а если в первой отсутствует - то из второй
```
Comment (blinkov) [inline ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md:412]:
```suggestion
1. `left` - первая структура;
2. `right` - вторая структура;
3. `mergeLambda` - _(опционально)_ позволяет задать функцию для объединения полей (аргументы: имя поля, Optional значение поля в первой структуре, Optional значение поля во второй структуре); по умолчанию выбирается значение поля из первой структуры, а если в первой отсутствует — из второй.
```

File: ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md [hunk 4]
```diff
@@ -392,3 +392,39 @@ ForceSpreadMembers(list_of_tuples:List<Tuple<String, T>>, result_keys:List<Strin
...
+
+**Примеры**
+``` yql
+$merge = ($name, $l, $r) -> { return Coalesce($l, 0) + Coalesce($r, 0); };
```
Comment (blinkov) [inline ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md:416]:
```suggestion
$merge = ($name, $l, $r) -> {
    return ($l ?? 0) + ($r ?? 0);
};
```

**Review by blinkov — DISMISSED** [review]

File: ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md [hunk 5]
```diff
@@ -392,3 +392,39 @@ ForceSpreadMembers(list_of_tuples:List<Tuple<String, T>>, result_keys:List<Strin
 SELECT
   ForceSpreadMembers([('a',1),('a',2),('c',100)],['a','b']); -- (a: 2, b: null)
 ```
+
+## StructUnion, StructIntersection, StructDifference, StructSymmetricDifference
```
Comment (blinkov) [inline ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md:396]: плюс перевод на английский

Comment (avevad) [reply ydb/docs/ru/core/yql/reference/yql-core/builtins/_includes/struct.md:396]: добавил

File: ydb/docs/en/core/yql/reference/yql-core/builtins/_includes/struct.md [hunk 6]
```diff
@@ -318,3 +318,42 @@ SELECT
   ForceSpreadMembers([('a',1),('a',2),('c',100)],['a','b']); -- (a: 2, b: null)
 ```
 
+## StructUnion, StructIntersection, StructDifference, StructSymmetricDifference
+
+Combine two structures using one of the four methods (using the provided lambda to merge fields with same name):
```
Comment (blinkov) [inline ydb/docs/en/core/yql/reference/yql-core/builtins/_includes/struct.md:323]:
```suggestion
Combine two structures using one of the four methods (using the provided lambda to merge fields with the same name):
```

File: ydb/docs/en/core/yql/reference/yql-core/builtins/_includes/struct.md [hunk 7]
```diff
@@ -318,3 +318,42 @@ SELECT
...
+
+* `StructUnion` adds all fields of both of the structures to the result;
+* `StructIntersection` — only the fields which are present in the both of the structures;
+* `StructDifference` — only the fields of `left`, which are absent in `right`;
+* `StructSymmetricDifference` — all fields which are present in exactly one of the structures.
```
Comment (blinkov) [inline ydb/docs/en/core/yql/reference/yql-core/builtins/_includes/struct.md:328]:
```suggestion
* `StructUnion` adds all fields of both of the structures to the result.
* `StructIntersection` adds only the fields which are present in both of the structures.
* `StructDifference` adds only the fields of `left`, which are absent in `right`.
* `StructSymmetricDifference` adds all fields that are present in exactly one of the structures.
```

File: ydb/docs/en/core/yql/reference/yql-core/builtins/_includes/struct.md [hunk 8]
```diff
@@ -318,3 +318,42 @@ SELECT
...
+StructIntersection(left:Struct<...>, right:Struct<...>[, mergeLambda:(name:String, l:T1?, r:T2?)->T])->Struct<...>
+StructDifference(left:Struct<...>, right:Struct<...>)->Struct<...>
+StructSymmetricDifference(left:Struct<...>, right:Struct<...>)->Struct<...>
+```
+
+Arguments:
+
+1. `left` - first structure;
+2. `right` - second structure;
+3. `mergeLambda` - _(optional)_ function to merge fields with same name (arguments: field name, Optional field value of the first struct, Optional field value of the second struct); by default, the first structure's field value is used, if present, and the second one's in other cases
```
Comment (blinkov) [inline ydb/docs/en/core/yql/reference/yql-core/builtins/_includes/struct.md:342]:
```suggestion
Arguments:

1. `left` - first structure.
2. `right` - second structure.
3. `mergeLambda` - _(optional)_ function to merge fields with the same name (arguments: field name, Optional field value of the first struct, Optional field value of the second struct). By default, if present, the first structure's field value is used, and the second one's in other cases.
```

**Review by fomichev3000 — APPROVED** [review]

**Pull request merged**